plugins/linux-v4l2/linux-v4l2.c

```diff
@@ -15,18 +15,35 @@ You should have received a copy of the GNU General Public License
 along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 #include <obs-module.h>
+#include <util/platform.h>
 
 OBS_DECLARE_MODULE()
 OBS_MODULE_USE_DEFAULT_LOCALE("linux-v4l2", "en-US")
 MODULE_EXPORT const char *obs_module_description(void)
 {
-	return "Video4Linux2(V4L2) sources";
+	return "Video4Linux2(V4L2) sources/virtual camera";
 }
 
 extern struct obs_source_info v4l2_input;
+extern struct obs_output_info virtualcam_info;
 
 bool obs_module_load(void)
 {
 	obs_register_source(&v4l2_input);
+
+	obs_data_t *obs_settings = obs_data_create();
+
+	if (os_file_exists("/var/lib/dkms/v4l2loopback")) {
```

Comment:
I remember specifically not using dkms *years ago* on Gentoo, back in the days when I built my own kernels...  But maybe *everyone* uses it now.

Anyway, no need to hold up merging over this: someone can always try to improve it later right?

Reply:
@cbm755 is correct that this won't work for anyone who builds the kernel module regularly. There's no need for DKMS to build kernel modules. This is where it lives for my 5.4 kernel for example `/lib/modules/5.4.60/video/v4l2loopback.ko`.
As @cbm755 mentioned the easiest alternative is to check modinfo, that should work in all cases
```
$ modinfo v4l2loopback 
filename:       /lib/modules/5.4.60/video/v4l2loopback.ko
license:        GPL
author:         Vasily Levin, IOhannes m zmoelnig <zmoelnig@iem.at>,Stefan Diewald,Anton Novikovet al.
description:    V4L2 loopback video device
depends:        videodev
retpoline:      Y
name:           v4l2loopback
vermagic:       5.4.60 SMP mod_unload 
parm:           debug:debugging level (higher values == more verbose) (int)
parm:           max_buffers:how many buffers should be allocated (int)
parm:           max_openers:how many users can open loopback device (int)
parm:           devices:how many devices should be created (int)
parm:           video_nr:video device numbers (-1=auto, 0=/dev/video0, etc.) (array of int)
parm:           card_label:card labels for every device (array of charp)
parm:           exclusive_caps:whether to announce OUTPUT/CAPTURE capabilities exclusively or not (array of bool)
parm:           max_width:maximum frame width (int)
parm:           max_height:maximum frame height (int)
$ echo $?
0
$ modinfo v4l2loopback1
modinfo: ERROR: Module v4l2loopback1 not found.
$ echo $?              
1
```

P.S. Tbh I don't think a userspace/end user application should be loading modules but as long as one can load the module upfront and OBS doesn't do anything it's OK for my use.

Reply:
@cg2121 are you maintaining this branch?  Shall we send pull requests to this branch on your fork?

I agree with @simonvanderveldt: maybe a compromise would be a dialog something like:
```
 Cannot start a virtual camera via a loopback video device.

 The v4l2loopback kernel module must be loaded.  You can load it yourself
 or we can try to load it for you (you'll be prompted for a password).

 [cancel]  [try again]  [try to load it automatically]
```
After seeing some of the stuff @CMCDragonkai is doing, I think we also need allow the UI to choose the video device...

Reply:
This simply won't work places. It flat out won't work in NixOS, ever.

Reply:
One could check the existence of `/sys/module/v4l2loopback/parameters/video_nr` to verify that the module is loaded correctly. This will work independently of how the kernel module is installed, i.e. also works for those not using DKMS.

```suggestion
	if (os_file_exists("/sys/module/v4l2loopback/parameters/video_nr")) {
```

This file contains the video device numbers associated with the v4l2loopback, e.g. on my system:

```
7,8,-1,-1,-1,-1,-1,-1
```

Output of `v4l2-ctl --list-devices`:

```
tablet (platform:v4l2loopback-000):
	/dev/video7

zoom (platform:v4l2loopback-001):
	/dev/video8

Integrated_Webcam_HD: Integrate (usb-0000:00:14.0-5):
	/dev/video0
	/dev/video1
	/dev/video2
	/dev/video3

B525 HD Webcam (usb-0000:3f:00.0-1.4):
	/dev/video4
	/dev/video5
```

Reply:
+1 for `modinfo`. On OpenSUSE `v4l2loopback` is built as a separate kernel module and accessible via `modinfo`/`modprobe`, no DKMS involved.

Reply:
The latest update of this PR now uses modinfo to determine if the module is installed.